Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Account.hs

```diff
@@ -0,0 +1,113 @@
+  )
+import qualified Storage.Queries.Role as QRole
+import Tools.Auth.Api
+import Tools.Auth.Merchant
+import "lib-dashboard" Tools.Error
+  ( GenericError (InvalidRequest),
+    PersonError (PersonDoesNotExist),
+    RoleError (RoleDoesNotExist),
+  )
+
+getAccountFetchUnverifiedAccounts ::
+  Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  ApiTokenInfo ->
+  Kernel.Prelude.Maybe Kernel.Prelude.UTCTime ->
+  Kernel.Prelude.Maybe Kernel.Prelude.UTCTime ->
+  Kernel.Prelude.Maybe Kernel.Prelude.Text ->
+  Kernel.Prelude.Maybe Common.FleetOwnerStatus ->
+  Kernel.Prelude.Maybe Kernel.Prelude.Int ->
+  Kernel.Prelude.Maybe Kernel.Prelude.Int ->
+  Environment.Flow [Common.PersonAPIEntity]
+getAccountFetchUnverifiedAccounts _merchantShortId _opCity _apiTokenInfo mbFromDate mbToDate mbMobileNumber mbStatus mbLimit mbOffset = do
+  encryptPersonLs <- findAllByFromDateAndToDateAndMobileNumberAndStatusWithLimitOffset mbFromDate mbToDate mbMobileNumber mbStatus mbLimit mbOffset
+  traverse convertPersonToPersonAPIEntity encryptPersonLs
+  where
+    convertPersonToPersonAPIEntity DP.Person {..} = do
+      role <- QRole.findById roleId >>= fromMaybeM (RoleDoesNotExist roleId.getId)
+      mobileNumber' <- decrypt mobileNumber
+      email' <- traverse decrypt email
+      pure $
+        Common.PersonAPIEntity
+          { id = Kernel.Types.Id.cast id,
+            roleAPIEntity = convertRoleToRoleAPIEntity role,
+            email = email',
+            mobileNumber = mobileNumber',
+            dashboardAccessType = castDashboardAccessType <$> dashboardAccessType,
+            ..
+          }
+    convertRoleToRoleAPIEntity DRole.Role {..} =
+      Common.RoleAPIEntity
+        { id = Kernel.Types.Id.cast id,
+          name = name,
+          dashboardAccessType = castDashboardAccessType dashboardAccessType,
+          description = description
+        }
+
+castDashboardAccessType :: DRole.DashboardAccessType -> Common.DashboardAccessType
+castDashboardAccessType = \case
+  DRole.DASHBOARD_USER -> Common.DASHBOARD_USER
+  DRole.DASHBOARD_ADMIN -> Common.DASHBOARD_ADMIN
+  DRole.FLEET_OWNER -> Common.FLEET_OWNER
+  DRole.DASHBOARD_RELEASE_ADMIN -> Common.DASHBOARD_RELEASE_ADMIN
+  DRole.MERCHANT_ADMIN -> Common.MERCHANT_ADMIN
+  DRole.RENTAL_FLEET_OWNER -> Common.RENTAL_FLEET_OWNER
+  DRole.MERCHANT_MAKER -> Common.MERCHANT_MAKER
+  DRole.MERCHANT_SERVER -> Common.MERCHANT_SERVER
+  DRole.DASHBOARD_OPERATOR -> Common.DASHBOARD_OPERATOR
+
+postAccountVerifyAccount ::
+  Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  ApiTokenInfo ->
+  Common.VerifyAccountReq ->
+  Environment.Flow Kernel.Types.APISuccess.APISuccess
+postAccountVerifyAccount merchantShortId opCity apiTokenInfo req = do
+  let personId = Kernel.Types.Id.cast req.fleetOwnerId
+  case req.status of
+    Common.Rejected ->
+      softDeletePerson personId req.reason
```

Comment:
here, I think would be useful to clean up existing tokens, just in case person was previously approved
See `Domain.Action.Dashboard.Registration.logout` function

Reply:
Also, lets call bpp here the same as for `Approved` case, so we can update `enabled` on bpp side